src/atoms/affine/kron.jl

```diff
@@ -23,7 +23,11 @@ function kron(a::AbstractExpr, b::Value)
     for i in 1:size(a)[1]
         row = AbstractExpr[]
         for j in 1:size(a)[2]
-            push!(row, a[i, j] .* b)
+            if isreal(b)
+                push!(row, a[i, j] * real(b))
```

Comment:
You should just be able to use this definition only and ditch the conditional, since `real` is a no-op for real inputs.

Reply:
If `b` is complex, we don't want to take the real part though. We just don't want the type to be complex if the entries of `b` are all real, otherwise it complains later. (Not my fix, just copying the workaround from the other method).

Reply:
Oh sorry, I read the logic backwards for some reason.